### PR TITLE
adding first unittest part

### PR DESCRIPTION
**Pull request by wojtekesz**
intro for activities unit test
include parse_to_utc and convert_to_utc methods
does not include post and get methods that require mock (probably)

### CODE REVIEW

File: api_backend_server/tests/unit_tests/pytest_activities_apis.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+from api_backend_server.web_app.apis.activities import _convert_unix_to_iso8610, _parse_to_utc_iso8610
```
Comment (thegreatmoonmoon) [inline api_backend_server/tests/unit_tests/pytest_activities_apis.py]: Name of test modules needs to start with `test_`. Can we change the file name from:
`pytest_activities_apis.py`
to:
`test_activities_apis.py`
?

File: api_backend_server/tests/unit_tests/pytest_activities_apis.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+from api_backend_server.web_app.apis.activities import _convert_unix_to_iso8610, _parse_to_utc_iso8610
+from api_backend_server.web_app.apis.activities import create_activity, get_user_activities
```
Comment (thegreatmoonmoon) [inline api_backend_server/tests/unit_tests/pytest_activities_apis.py]: for long imports ( > 88 characters) we can write them like that:
```python
from api_backend_server.web_app.apis.activities import (
    _convert_unix_to_iso8610,
    _parse_to_utc_iso8610,
)
```

Or, if we want all attributes to be imported to the test, a wild import in this case will also be acceptable:
```python
from api_backend_server.web_app.apis.activities import *
```

File: api_backend_server/tests/unit_tests/pytest_activities_apis.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from api_backend_server.web_app.apis.activities import _convert_unix_to_iso8610, _parse_to_utc_iso8610
+from api_backend_server.web_app.apis.activities import create_activity, get_user_activities
+import pytest
+import datetime
+import mock as mock
```
Comment (thegreatmoonmoon) [inline api_backend_server/tests/unit_tests/pytest_activities_apis.py]: mocks are not used in the test at the moment - let's not import them if not necessary :dog:

File: api_backend_server/tests/unit_tests/pytest_activities_apis.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+from api_backend_server.web_app.apis.activities import _convert_unix_to_iso8610, _parse_to_utc_iso8610
+from api_backend_server.web_app.apis.activities import create_activity, get_user_activities
+import pytest
+import datetime
+import mock as mock
+from mock import MagicMock
+
+
+def test__parse_to_utc_iso8610_should_succeed():
```
Comment (thegreatmoonmoon) [inline api_backend_server/tests/unit_tests/pytest_activities_apis.py]: Test fails - you are comparing a datetime object with a string, so assert will fail :stuck_out_tongue:

File: api_backend_server/tests/unit_tests/pytest_activities_apis.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    # when
+    result = _parse_to_utc_iso8610(string_timestamp)
+
+    # then
+    assert result == '2020-04-30 07:34:35.414677+00:00'
+
+
+def test__parse_to_utc_iso8610_should_throw_exception():
+    # given
```
Comment (thegreatmoonmoon) [inline api_backend_server/tests/unit_tests/pytest_activities_apis.py]: Those comments (given, when, then) are unnecessary imo for this test (may be useful for the tests that conform to that pattern though)

File: api_backend_server/tests/unit_tests/pytest_activities_apis.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    unix_timestamp  = 1588248502
+
+    # when
+    with pytest.raises(Exception):
+        _convert_unix_to_iso8610(unix_timestamp)
+    
+    # then
+
+# NOT FINISHED TEST METHOD
```
Comment (thegreatmoonmoon) [inline api_backend_server/tests/unit_tests/pytest_activities_apis.py]: We shouldn't commit commented out blocks of code - you could use a pytest fixture [skip](https://docs.pytest.org/en/latest/reference.html#pytest-skip) to temporarily skip unfinished test

**Review by thegreatmoonmoon — CHANGES_REQUESTED** [review]
@wojtekesz Have you tried to run those tests with pytest? they fail unfortunately :smile:

**Review by Qamas95 — APPROVED** [review]
:octocat:

**Review by thegreatmoonmoon — APPROVED** [review]
:field_hockey: fixed

**Pull request merged**